app/src/main/java/com/haeti/sopose/navigation/MainNavHost.kt

```diff
@@ -1,39 +1,119 @@
 
+import android.annotation.SuppressLint
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import androidx.navigation.NavGraphBuilder
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.navigation
 import com.haeti.sopose.android.AndroidScreen
 import com.haeti.sopose.auth.AuthViewModel
 import com.haeti.sopose.home.HomeScreen
+import com.haeti.sopose.home.HomeViewModel
 import com.haeti.sopose.login.LoginScreen
 import com.haeti.sopose.mypage.MypageScreen
 import com.haeti.sopose.signup.SignUpScreen
 
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
 fun MainNavHost(
     navController: NavHostController,
     viewModel: AuthViewModel
 ) {
-    NavHost(
-        navController = navController,
-        startDestination = Screen.Login.route
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    val currentRoute = navBackStackEntry?.destination?.route
+    val homeViewModel: HomeViewModel = hiltViewModel()
+
+    Scaffold(
+        topBar = {
+            when (currentRoute) {
+                Screen.Login.route -> {
+                    CenterAlignedTopAppBar(title = {
+                        Text(
+                            text = "Welcome to Sopose",
+                            style = MaterialTheme.typography.headlineLarge
+                        )
+                    })
+                }
+
+                Screen.SignUp.route -> {
+                    CenterAlignedTopAppBar(title = {
+                        Text(
+                            text = "Sign Up",
+                            style = MaterialTheme.typography.headlineLarge,
+                        )
+                    })
+                }
+            }
+        },
+        bottomBar = {
+            if (BottomNavItem.isInBottomNav(currentRoute)) {
+                BottomNavigationBar(navController = navController)
+            }
+        }
     ) {
+
+        NavHost(
+            navController = navController,
+            startDestination = NavGraph.Auth.route
+        ) {
+            authGraph(
+                navController = navController,
+                authViewModel = viewModel,
+                modifier = Modifier.padding(it)
+            )
+            mainGraph(authViewModel = viewModel, homeViewModel)
```

Comment:
이런 사소한거 밖에 꼬투리잡을게 없네요...
```suggestion
            mainGraph(
                authViewModel = viewModel, 
                homemViewModel = homeViewModel
            )
```